plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/encoder/protobuf/ProtobufSchemaParser.java

```diff
@@ -95,21 +98,24 @@ private Type getType(FieldDescriptor fieldDescriptor)
         return baseType;
     }
 
-    private Type getTypeForMessage(FieldDescriptor fieldDescriptor)
+    private Type getTypeForMessage(FieldDescriptor fieldDescriptor, Set<String> processedMessages)
     {
         Descriptor descriptor = fieldDescriptor.getMessageType();
-        if (fieldDescriptor.getMessageType().getFullName().equals(TIMESTAMP_TYPE_NAME)) {
+        if (descriptor.getFullName().equals(TIMESTAMP_TYPE_NAME)) {
             return createTimestampType(6);
         }
+        if (!processedMessages.add(descriptor.getFullName())) {
+            return UnknownType.UNKNOWN;
```

Comment:
alternatively we can inject `TypeManager` and do `typeManager.getType(new TypeSignature("unknown"))`.

Reply:
Alternatively you can throw a TrinoException with relevant message.

Reply:
done: https://github.com/trinodb/trino/pull/15312/files#diff-e59ff6a36068bd7da2bc7a4c4996a696ddac09a449542a981673cafee5643172R166